### PR TITLE
#42 api gateway registry integration

### PR DESCRIPTION
**Pull request by Moses-odessa**
connect to #42

### CODE REVIEW

File: src/main/java/ua/com/juja/microservices/teams/controller/TeamController.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
     @Inject
     private TeamService teamService;
 
-    @PostMapping(value = "${teams.endpoint.activateTeam}", consumes = "application/json", produces = "application/json")
+    @PostMapping(value = "/v1/teams", consumes = "application/json", produces = "application/json")
```
Comment (adaleorion) [inline src/main/java/ua/com/juja/microservices/teams/controller/TeamController.java]: I think v1 should be in controller mapping

File: src/main/java/ua/com/juja/microservices/teams/service/TeamService.java [hunk 2]
```diff
@@ -29,16 +28,14 @@
 public class TeamService {
 
     private static final int TEAM_SIZE = 4;
+    private final String teamsDirection = "teams";
```
Comment (adaleorion) [inline src/main/java/ua/com/juja/microservices/teams/service/TeamService.java]: Why do you use variable value instead property in configuration file ?

File: src/test/java/ua/com/juja/microservices/teams/service/TeamServiceTest.java [hunk 3]
```diff
@@ -41,6 +41,7 @@
 /**
  * @author Ivan Shapovalov
  * @author Andrii Sidun
+ * @author Vladimir Zadorozhniy
  */
 @RunWith(SpringRunner.class)
 @WebMvcTest(TeamService.class)
```
Comment (adaleorion) [inline src/test/java/ua/com/juja/microservices/teams/service/TeamServiceTest.java]: I think you should use `@SpringBootTest` annotation.

We disscus about this case on last meeting

https://github.com/JujaLabs/docs/wiki/%5BMFU%5D-BackEnd-Meeting-2017.12.01

File: src/test/java/ua/com/juja/microservices/teams/service/KeeperServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Vladimir Zadorozhniy
+ */
+@RunWith(SpringRunner.class)
+@WebMvcTest(KeeperService.class)
```
Comment (adaleorion) [inline src/test/java/ua/com/juja/microservices/teams/service/KeeperServiceTest.java]: I think you should use @SpringBootTest annotation.

We disscus about this case on last meeting
https://github.com/JujaLabs/docs/wiki/%5BMFU%5D-BackEnd-Meeting-2017.12.01

File: src/main/java/ua/com/juja/microservices/teams/utils/Utils.java [hunk 5]
```diff
@@ -1,28 +1,37 @@
...
-            return new ApiErrorMessage(
-                    500, "BotInternalError",
-                    "I'm, sorry. I cannot parse api error message from remote service :(",
-                    "Cannot parse api error message from remote service",
-                    e.getMessage(),
-                    Collections.singletonList(httpClientErrorException.getMessage())
-            );
+    public static ApiErrorMessage convertToApiError(String message) {
+        int contentExists = message.indexOf("content:");
+        ApiErrorMessage apiError = new ApiErrorMessage(
```
Comment (adaleorion) [inline src/main/java/ua/com/juja/microservices/teams/utils/Utils.java:16]: I think you should think about some code duplication when you create apiError. 
Maybe you need to extract some method for this case or change logic

**Review by adaleorion — COMMENTED** [review]
@Moses-odessa 
I need your attention to some my comments.
Please check and do give me your responses

File: src/main/java/ua/com/juja/microservices/teams/service/TeamService.java (hunk 2 above)
Comment (ivshapovalov) [reply src/main/java/ua/com/juja/microservices/teams/service/TeamService.java]: Exactly. We decided to remove url from properties, but not any variables

**Review by adaleorion — APPROVED** [review]

**Pull request merged**